Sources/AwaitKit.swift

```diff
@@ -73,5 +95,62 @@ public func await<T>(_ body: @escaping () throws -> T) throws -> T {
  */
 @discardableResult
 public func await<T>(_ promise: Promise<T>) throws -> T {
-  return try AwaitKit.awaitQueue.await(promise)
+    return try await(promise, queue: awaitQueue)
+}
+
+/**
+ Awaits that the given closure finished on the receiver and returns its value or throws an error if the closure failed.
+ 
+ - parameter body: The closure that is executed on the receiver.
+ - throws: The error sent by the closure.
+ - returns: The value of the closure when it is done.
+ - seeAlso: await(promise:)
+ */
+@discardableResult
+public func await<T>(_ body: @escaping () throws -> T, queue: DispatchQueue) throws -> T {
```

Comment:
The `AwaitKit` v1 used the same API signature. But I think this kind of method signature does not conform to the [Swift3 API design guidelines](https://swift.org/documentation/api-design-guidelines/).

As proposed in the issue #9, I prefer to add a category to have something similar to:

```swift
DispatchQueue.main.ak.async {}
// or
DispatchQueue.main.ak_async {}
```